### ISSUE TITLE
Port twisted.protocols.loopback to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6109]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6109|
|Type|enhancement|
|Created|2012-10-15 15:57:31Z|
|Branch|https://github.com/twisted/twisted/tree/loopback-py3-6109-2|

Some HTTP tests use it, so it would at least help get full test coverage for Twisted Web.

<details><summary>Searchable metadata</summary>

```
trac-id__6109 6109
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_loopback_py3_6109_2 branches/loopback-py3-6109-2
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1350316651000000 1350316651000000
changetime__1350403725000000 1350403725000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

SSL unit tests also need this, so I guess it is part of the minimal subset.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [36183]) Branching to 'loopback-py3-6109'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst<br>@exarkun set status to `new`|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/loopback-py3-6109).  Test coverage for `loopbackAsync` seems pretty good, and that's all we care about I think.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set owner to @exarkun|
|-|-|

Thanks!

  1. `LoopbackRelay` has a bunch of strings that need to be converted to bytes.
  1. Open a ticket for missing tests.

Fix, then merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [36199]) Branching to 'loopback-py3-6109-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

Thanks!

> LoopbackRelay has a bunch of strings that need to be converted to bytes.

Moved the module to the not-quite-ported list instead - r36201.

> Open a ticket for missing tests.

Opened [#6138](https://github.com/twisted/twisted/issues/6138)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [36203]) Merge loopback-py3-6109-2

Author: exarkun
Reviewer: itamarst
Fixes: [#6109](https://github.com/twisted/twisted/issues/6109)

Port `twisted.protocols.loopback.loopbackAsync` to Python 3.
